src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/BackgroundAspectEffect/BackgroundAspectEffectRouter.ios.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using UIKit;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+using Effects = Xamarin.CommunityToolkit.iOS.Effects;
+
+[assembly: ExportEffect(typeof(Effects.BackgroundAspectEffectRouter), nameof(BackgroundAspectEffectRouter))]
+
+namespace Xamarin.CommunityToolkit.iOS.Effects
+{
+	public class BackgroundAspectEffectRouter : PlatformEffect
+	{
+		UIImageView? imageView;
+
+		CancellationTokenSource cancellationTokenSource;
+
+		public BackgroundAspectEffectRouter()
+		{
+			cancellationTokenSource = new CancellationTokenSource();
+		}
+
+		protected override void OnAttached()
+			=> Task.Run(ApplyBackgroundImageAsync, cancellationTokenSource.Token);
```

Comment:
Does this need to be a `Task.Run`? Why not call the Async method directly?

Reply:
So that it can be cancelled if the `OnElementProperty` fires in quick succession